docs/design-documents/README.md

```diff
@@ -0,0 +1,40 @@
+* Your design document itself should also be named with the same name of the as the folder, so that it serves
+  as your master design document, if you happen to use other .md files under same folder as annexures or reference.
+* The diagrams you are using may exist in the sub-folder *diagrams* under the new folder you have created.
+  See [Example Design Document](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/example_feature_design/example_feature_design.md) for example design document.
+
+### Using diagrams in Design documents
+Its highly encouraged to create diagrams to depict your software design. The [Design Document Template]((https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/design_template.md) provided here uses diagrams generated using draw.io. draw.io generates XMLs for each diagram and those corresponding XMLs are also provided under the diagrams folder, and you are free to use that as your reference for generating your own diagrams. And, when you are generating your own diagrams it would be better to provide the metadata/other files(e.g, if you are using draw.io provide XMLs from draw.io) required for future editing of those diagrams.
+
+## Design Process
+* Developer generates a design document located under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}* and generates a pull-request to Mbed-OS.
+  * NOTE: If you are part of an organization and would like to do design review within your organization before landing the pull-request, its ok to do so.
+  * The pull request may also include the initial code implementation. It's upto to the develper to decide if the initial implementation comes along with the design document or later.
+  * Although, if you are planning a large change, its encoraged to start the design review early before the implementation. This also help to avoid major re-work of your implementation if required by the design review.
+* Design review follows the same process as code review. Please refer contribution process as outlined in [Contributing to Mbed-OS](https://os.mbed.com/contributing/) for more information.
+* Once the design is approved, implementation is completed/adjusted according to the latest design document. Developer can create the pull-request with the final implementation, if its coming separately.
+
+## Capturing future changes to design
+If your implementation changes in future, corresponsding design documents may be updated to capture the new design.
+It's possible that a new implementation may require changes to some previous implementations. In that case, capture
+the new changes in the original design document as well. Also create references to other(or older) design documents in the
+References section for the new design document.
+
```

Comment:
I comment here but it comes after design phase - what about the implementation ? Is author expected to implement, or rather anyone can do, etc. I think a section here would be sufficient about "after design doc is accepted"

Reply:
No really, its possible that design document author could be different from implementer. In fact, that's where the document really helps. If the document is detailed enough, anyone can take the document and go do the implementation. Its not absolutely necessary that implementation should be started only after design document review. The only guideline here is if there is change to design it will impact your implementation as well, and thus for large features where lot of design feedbacks are expected it's advisable to start your implementation once major feedbacks are captured. Again, its up to the implementer to make that judgement depending on how big/which part of mbed-os the feature is. For example, API designs can be very intensive and may drastically change over the period of review. In those cases, its best to delay your major implementation cycle until that agreement is reached. But for localized standard algorithms implementations not impacting other parts of the system/application writers it may be ok to have an implementation initially.